chesster.js

```diff
@@ -617,446 +259,128 @@ controller.hears([
-    });
-}
-
-function getRatingString(rating){
-    return ( rating ? " (" + rating + ")" : "" );
-}
-
-function formatPairingResult(details, callback){
-    var localTime = moment.utc(details.date + " " + details.time, "MM/DD HH:mm").utcOffset(details.tz_offset);
-    var localDateTimeString = localTime.format("dddd [at] HH:mm");
-    var player = details.player;
-    var opponent = details.opponent;
-    var color = details.color;
-    var rating = details.rating;
-
-    if (!localTime.isValid()) {
-        callback(player + " will play as " + color +" against " + opponent + getRatingString(rating) + ".  The game is unscheduled.");
-    } else if (moment.utc().isAfter(localTime)) {
-        // If the match took place in the past, display the date instead of the day
-        localDateTimeString = localTime.format("MM/DD [at] HH:mm");
-        callback(player + " played as " + color +" against " + opponent + getRatingString(rating) + " on " + localDateTimeString + ".");
-    } else {
-        // Otherwise display the time until the match
-        var timeUntil = localTime.fromNow(true);
-        callback(player + " will play as " + color +" against " + opponent + getRatingString(rating) + " on " + localDateTimeString + " which is in " + timeUntil + ".");
-    }
-}
-
-function parsePairingResult(player, tz_offset, opponent, color, date, time, callback) {
-
-    var opponentName = uncaptain(opponent);
-    getPlayerByName(opponentName, function(error, opponent) {
-        if(opponent){
-            getClassicalRating(opponent, function(rating) {
-                formatPairingResult({
-                    "player": player, 
-                    "tz_offset": tz_offset, 
-                    "opponent": opponentName, 
-                    "color": color, 
-                    "date": date, 
-                    "time": time, 
-                    "rating": rating, 
-                }, callback);
-            });
-        }else{
-            console.error(JSON.stringify(error));
-            formatPairingResult({
-                "player": player,
-                "tz_offset": tz_offset,
-                "opponent": opponentName,
-                "color": color,
-                "date": date,
-                "time": time,
-            }, callback);
-        }
-    });
-}
-
-/* standings */
-
-function prepareStandingsMessage(){
-    return "Here is the standings sheet:\n" + 
-            config.links.team + 
-            "\nAlternatively, try [ @chesster result <competitor> <round> ] - coming soon...";
-    
-}
-
-
-function sayStandings(convo){
-    convo.say(prepareStandingsMessage());
-}
-
-controller.hears([
-    'standings'
-],[
-    'direct_mention', 
-    'direct_message'
-],function(bot, message) {
-    bot_exception_handler(bot, message, function(){
-        bot.reply(message, prepareStandingsMessage());
+chesster.hears({
+    middleware: [slack.requiresLeague, slack.requiresModerator],
```

Comment:
Yeah, being able to quickly write commands that require a moderator is pretty awesome.